src/NetMQ/zmq/Transports/Tcp/TcpConnector.cs

```diff
@@ -180,20 +189,22 @@ private void StartConnecting()
         }
 
         /// <summary>
-        /// 
+        /// This method is called when a message Send operation has been completed.
         /// </summary>
-        /// <param name="socketError"></param>
-        /// <param name="bytesTransferred"></param>
-        /// <exception cref="NetMQException">a non-recoverable socket error occurred.</exception>
+        /// <param name="socketError">a SocketError value that indicates whether Success or an error occurred</param>
+        /// <param name="bytesTransferred">the number of bytes that were transferred</param>
+        /// <exception cref="NetMQException">A non-recoverable socket error occurred.</exception>
         public void OutCompleted(SocketError socketError, int bytesTransferred)
         {
+            m_ioObject.RemoveSocket(m_s);
+            m_isHandleValid = false;
+
             if (socketError != SocketError.Success)
             {
-                m_ioObject.RemoveSocket(m_s);
```

Comment:
Just checked - yes, that's what was in your master-branch, which I merged into my edited code before commiting it, attempting to reduce your work in doing this merge.  It doe NOT look like correct code!  Why mark the handle as invalid, unless the socketError is indeed != Success  ?!   How did this happen?